src/pages/NodesTelemetry.vue

```diff
@@ -0,0 +1,383 @@
+  }
+);
+
+useIntervalFn(getNetworkMetrics, 15000, {
+  immediateCallback: true,
+});
+useIntervalFn(getPeersInfo, 30000, {
+  immediateCallback: true,
+});
+
+function formatTimeSpan(date1: Date | null, date2: Date | null) {
+  if (!date1 || !date2) return '-';
+
+  return (date1.getTime() - date2.getTime()) / 1000 + 's';
+}
+</script>
+
+<template>
+  <div class="nodes-telemetry-page">
+    <BaseLoading
+      v-if="isMetricsLoading && !metrics"
+      class="nodes-telemetry-page_loading"
+    />
+    <div
+      v-else-if="metrics"
+      class="nodes-telemetry-page__stats"
+    >
+      <div class="nodes-telemetry-page__stats-stat">
+        <span class="nodes-telemetry-page__stats-stat-value">#{{ formatNumber(metrics.latest_block) }}</span>
+        <span class="nodes-telemetry-page__stats-stat-label">{{ $t('telemetry.bestBlock') }}</span>
+      </div>
+      <div class="nodes-telemetry-page__stats-stat">
+        <span class="nodes-telemetry-page__stats-stat-value">#{{ formatNumber(metrics.finalized_block) }}</span>
+        <span class="nodes-telemetry-page__stats-stat-label">{{ $t('telemetry.finalizedBlock') }}</span>
+      </div>
+      <div class="nodes-telemetry-page__stats-stat">
+        <span class="nodes-telemetry-page__stats-stat-value">{{ Math.trunc(metrics.average_commit_time_ms) / 1000 }}s</span>
+        <span class="nodes-telemetry-page__stats-stat-label">{{ $t('telemetry.averageBlockTime') }}</span>
+      </div>
+      <div class="nodes-telemetry-page__stats-stat">
+        <span class="nodes-telemetry-page__stats-stat-value">{{ Math.trunc(metrics.average_block_time_ms) / 1000 }}s</span>
+        <span class="nodes-telemetry-page__stats-stat-label">{{ $t('telemetry.averageBlockCommitTime') }}</span>
+      </div>
+      <div class="nodes-telemetry-page__stats-stat">
+        <div class="nodes-telemetry-page__stats-stat-value nodes-telemetry-page__stats-stat-last-block">
+          <div v-if="metrics.latest_block_created_at">
+            <LatestBlock :date="metrics.latest_block_created_at" />
+            <span>s</span>
+          </div>
+          <div v-else>
+            Unknown
+          </div>
+        </div>
+        <span class="nodes-telemetry-page__stats-stat-label">{{ $t('telemetry.lastBlock') }}</span>
+      </div>
+    </div>
+    <BaseContentBlock
+      :title="$t('telemetry.nodes')"
+      class="nodes-telemetry-page__list"
+    >
+      <BaseTable
+        disable-pagination
+        :loading="streamStatus === 'CONNECTING'"
+        :items="Array.from(peers.values())"
+        container-class="nodes-telemetry-page__list-container"
+        :breakpoint="1440"
+      >
+        <template #header>
+          <div class="nodes-telemetry-page__list-row">
+            <span class="h-sm cell">{{ $t('telemetry.publicKey') }}</span>
+            <span class="h-sm cell">{{ $t('telemetry.publicUrl') }}</span>
+            <span class="h-sm cell">{{ $t('telemetry.block') }}</span>
```

Comment:
This could also be a link to the block details page.

Reply:
I see that you added it in the header, but not in the table.

I mean something like:
```suggestion
            <span class="h-sm cell"><BaseLink :to="...">{{ $t('telemetry.block') }}</BaseLink></span>
```

Reply:
My bad - this is the table's header...